### ISSUE TITLE
Custom Client Hello extension not appearing in client hello callback

### ISSUE THREAD

**Issue by miguelarman** [openssl/openssl#20560]
Hello. I am implementing a pair of TLSv1.3 server and client programs. The step I am currently working on is including a custom extension in the Client Hello message. For this I am using [SSL_CTX_add_custom_ext](https://www.openssl.org/docs/manmaster/man3/SSL_CTX_add_custom_ext.html) and the needed callbacks:
```
SSL_CTX_add_custom_ext(ctx, 65535, SSL_EXT_TLS1_3_ONLY | SSL_EXT_CLIENT_HELLO, custom_extension_add_callback, custom_extension_free_callback, NULL, custom_extension_parse_callback, NULL)
```

The created extension is being added, as I can see it via Wireshark: 
![image](https://user-images.githubusercontent.com/22821797/226645001-5fd2d837-8f5a-49a6-af14-ee84875cd3ac.png)

Plus, the `add` and `free` callbacks are being called, as I can see in the output of the program: (checking the documentation I am not completely sure whether the `parse` callback not being called shows a possible error or not.)

```
...
Custom extension add callback has been called!
Custom extension free callback has been called!
...
```

For my tests, I used several different `ext_type` values, both greater and smaller than 255 (distinction made in [RFC8446](https://www.rfc-editor.org/rfc/rfc8446#section-11)). All of them can be seen in my Wireshark captures, with some being reserved values and then are given a name different to "Unknown type".

My next step was to check the value of this extension in the server program. Namely, in the client_hello callback. This function I know is working correctly because in it I can get and check the values of some other extensions (and it was fixed with the help of some other issues I posted here). But my problem is that this custom extension is not included in the returned values of `SSL_client_hello_get1_extensions_present`, and neither returned if I call `SSL_client_hello_get0_ext` with the specific extension type set when adding it.

Lastly, I tried setting the context to `SSL_EXT_TLS1_3_ONLY | SSL_EXT_CLIENT_HELLO` and `SSL_EXT_CLIENT_HELLO` with no changes in the behaviour.

Thanks in advance for all the help you can give me!

**Comment by miguelarman** [comment]
Hello again.

I found #18286 which kind of shows my same problem. As my id is 65535 (not recognized by OpenSSL as it is not a default extension), I needed to call `SSL_CTX_add_custom_ext` both in my server and client code. This allows `SSL_client_hello_get1_extensions_present` to return the custom id, and also `SSL_client_hello_get0_ext` to get the extension data.

Thus, it was all my server program was lacking.

Maybe, as a suggestion, the documentation for `SSL_client_hello_get0_ext` could include some of the message from @mattcaswell in the second paragraph of [this comment](https://github.com/openssl/openssl/issues/18286#issuecomment-1123436664) which ultimately led me to fix this.

**Comment by nhorman** [comment]
marking as inactive, will be closed at the completion of 3.4 dev without further input
